R/load_survey.r

```diff
@@ -73,44 +72,37 @@ load_survey <- function(files, ...) {
     can_merge <- vapply(survey_files, function(x) {
       length(intersect(colnames(contact_data[[x]]), colnames(main_surveys[[type]]))) > 0
     }, TRUE)
-    merge_files <- names(can_merge[which(can_merge)])
+    merge_files <- names(can_merge[can_merge])
     while (length(merge_files) > 0) {
       merged_files <- NULL
       for (file in merge_files)
       {
         common_id <- intersect(colnames(contact_data[[file]]), colnames(main_surveys[[type]]))
 
-        ## try a merge to see whether it can be done uniquely
-        test_merge <- merge(
-          main_surveys[[type]], contact_data[[file]], by = common_id,
-          all.x = TRUE, allow.cartesian = TRUE
-        )
-
-        if (nrow(test_merge) == nrow(main_surveys[[type]])) {
-          ## check if all IDs can be merged in
-          unique_main_survey_ids <-
-            unique(main_surveys[[type]][, common_id, with = FALSE])
-          unique_additional_survey_ids <-
-            unique(contact_data[[file]][, common_id, with = FALSE])
-
-          id_overlap <- merge(
-            unique_main_survey_ids, unique_additional_survey_ids,
-            by = common_id
+        # is the id unique and can the merge be done uniquely?
+        if (anyDuplicated(main_surveys[[type]][, common_id, with = FALSE]) == 0) {
```

Comment:
```suggestion
        if (anyDuplicated(contact_data[[file]][, common_id, with = FALSE]) == 0) {
```

I don't think we require uniqueness in the main file to merge, i.e. we can have a successful merge with:

#### `main_surveys[[type]]`

<table>
<tr>
	<td> part_id
	<td> hh_id
<tr>
	<td> 1
	<td> 1
<tr>
	<td> 2
	<td> 1
<tr>
	<td> 3
	<td> 2
</table>

#### `contact_data[[file]]`

<table>
<tr>
	<td> hh_id
<tr>
	<td> 1
<tr>
	<td> 2
</table>

Reply:
There is another problem here: ids can not be unique but it doesn't matter if duplicated ids are absent from `main_surveys[[type]]` since they will be dropped.

Reply:
How about this for an alternative approach that just tries a single merge and, I think, addresses all the issues:
https://github.com/epiforecasts/socialmixr/blob/simplify-load_survey-2/R/load_survey.r
I think it should solve the memory issues but I haven't tested it thorougly (yet).

Reply:
Yes, I try to avoid `tryCatch()` as much as possible but it seems unavoidable here. Please do push your changes on this branch or open a new PR. I have a couple of minor comments